content/microservices/grpc.md

```diff
@@ -440,3 +442,68 @@ lotsOfGreetings(requestStream: any, callback: (err: unknown, value: HelloRespons
 ```
 
 Here we used the `callback` function to send the response once processing of the `requestStream` has been completed.
+
+#### gRPC Metadata
+
+gRPC metadata is additional data that may be useful during the processing of requests and responses, but is not part of the actual application data. Metadata may include authentication tokens, request identifiers and tags for monitoring purposes, and data information such as the number of records in a data set.
+
+To read the metadata in `@GrpcMethod()` use the second handler metadata argument, which returns the gRPC class `Metadata`.
+
+You can send metadata from `@GrpcMethod()` using the function `sendMetadata()` which is provided by the object `GrpcCall` in the third handler argument.
+
+```typescript
+@@filename(heroes.controller)
+@Controller()
+export class HeroesService {
+  @GrpcMethod()
+  findOne(data: HeroById, metadata: Metadata, call: ServerUnaryCall<any>): Hero {
+    const srvMetadata = new Metadata();
+    const items = [
+      { id: 1, name: 'John' },
+      { id: 2, name: 'Doe' },
+    ];
+
+    srvMetadata.add('Set-Cookie', 'yummy_cookie=choco');
+    call.sendMetadata(srvMetadata);
+
+    return items.find(({ id }) => id === data.id);
+  }
+}
+@@switch
+@Controller()
+export class HeroesService {
+  @GrpcMethod()
+  findOne(data, metadata, call) {
+    const srvMetadata = new Metadata();
+    const items = [
+      { id: 1, name: 'John' },
+      { id: 2, name: 'Doe' },
+    ];
+
+    srvMetadata.add('Set-Cookie', 'yummy_cookie=choco');
+    call.sendMetadata(srvMetadata);
+
+    return items.find(({ id }) => id === data.id);
+  }
+}
+```
```

Comment:
This looks great. Can we add a hint on where to `import` `Metadata` from [similar to here](https://docs.nestjs.com/graphql/resolvers#code-first-resolver) (line 157 of graphql/resovler-map, GitHub won't link the markdown file properly)

Reply:
done